runtime/gc_modron_startup/mmparse.cpp

```diff
@@ -559,18 +559,132 @@ xlpSubOptionsParser(J9JavaVM *vm, IDATA xlpIndex, XlpError *xlpError, UDATA *req
 	return xlpErrorState;
 }
 
+/**
+ * 
+ * Given the requested size of the large pages to be used, the method will verify that value is supported by the system. 
+ * If the value is not supported, the system will set the largest available size instead. The user will also be notified 
+ * of the altered value being used.
+ * 
+ */
+void
+verifyLargePageSize(J9JavaVM *vm, UDATA requestedPageSize, UDATA requestedPageFlags, MM_GCExtensions *extensions)
+{
+	UDATA pageSize = requestedPageSize;
+	UDATA pageFlags = requestedPageFlags;
+	BOOLEAN isRequestedSizeSupported = FALSE;
+	PORT_ACCESS_FROM_JAVAVM(vm);
+
+	IDATA result = j9vmem_find_valid_page_size(0, &pageSize, &pageFlags, &isRequestedSizeSupported);
+
+	/*
+	 * j9vmem_find_valid_page_size happened to be changed to always return 0
+	 * However formally the function type still be IDATA so assert if it returns anything else
+	 */
+	Assert_MM_true(0 == result);
+
+	extensions->requestedPageSize = pageSize;
+	extensions->requestedPageFlags = pageFlags;
+
+	if (!isRequestedSizeSupported) {
+		/* Print a message indicating requested page size is not supported and a different page size will be used */
+		const char *oldQualifier, *newQualifier;
+		const char *oldPageType = NULL;
+		const char *newPageType = NULL;
+		UDATA oldSize = requestedPageSize;
+		UDATA newSize = pageSize;
+		qualifiedSize(&oldSize, &oldQualifier);
+		qualifiedSize(&newSize, &newQualifier);
+		if (0 == (J9PORT_VMEM_PAGE_FLAG_NOT_USED & requestedPageFlags)) {
+			oldPageType = getPageTypeString(requestedPageFlags);
+		}
+		if (0 == (J9PORT_VMEM_PAGE_FLAG_NOT_USED & pageFlags)) {
+			newPageType = getPageTypeString(pageFlags);
+		}
+		if (NULL == oldPageType) {
+			if (NULL == newPageType) {
+				j9nls_printf(PORTLIB, J9NLS_INFO, J9NLS_GC_OPTIONS_LARGE_PAGE_SIZE_NOT_SUPPORTED, oldSize, oldQualifier, newSize, newQualifier);
+			} else {
+				j9nls_printf(PORTLIB, J9NLS_INFO, J9NLS_GC_OPTIONS_LARGE_PAGE_SIZE_NOT_SUPPORTED_WITH_NEW_PAGETYPE, oldSize, oldQualifier, newSize, newQualifier, newPageType);
+			}
+		} else {
+			if (NULL == newPageType) {
+				j9nls_printf(PORTLIB, J9NLS_INFO, J9NLS_GC_OPTIONS_LARGE_PAGE_SIZE_NOT_SUPPORTED_WITH_REQUESTED_PAGETYPE, oldSize, oldQualifier, oldPageType, newSize, newQualifier);
+			} else {
+				j9nls_printf(PORTLIB, J9NLS_INFO, J9NLS_GC_OPTIONS_LARGE_PAGE_SIZE_NOT_SUPPORTED_WITH_PAGETYPE, oldSize, oldQualifier, oldPageType, newSize, newQualifier, newPageType);
+			}
+		}
+	}
+}
+
+/**
+ * 
+ * Obtains and sets the largest page size available on the system. 
+ * Returns true if the operation was successful.
+ * 
+ */
+bool
+gcGetLargestPageSize(J9JavaVM *vm, MM_GCExtensions *extensions)
```

Comment:
Misleading name: this is not "largest" but "default large" or "preferred large". AIX largest page size is 16G but it is not preferred

Reply:
Right. Changed here https://github.com/eclipse/openj9/pull/7476/commits/d46ab1721594429a9cc3da27610e98302877642b